src/SurfaceSchemes/surface_setups.jl

```diff
@@ -0,0 +1,234 @@
+            θAndQFluxes(; θ_flux = θ_flux0 * weight, q_flux = q_flux0 * weight)
+        parameterization = MoninObukhov(z0; fluxes, ustar)
+        return SurfaceState(; parameterization, T, p, q_vap)
+    end
+    return surface_state
+end
+
+struct ARM_SGP end
+function (::ARM_SGP)(params)
+    FT = eltype(params)
+    θ = FT(299)
+    p = FT(97000)
+    q_vap = FT(0.0152)
+    z0 = FT(0)
+    t_data = FT[0, 4, 6.5, 7.5, 10, 12.5, 14.5] .* 3600
+    shf_data = FT[-30, 90, 140, 140, 100, -10, -10]
+    lhf_data = FT[5, 250, 450, 500, 420, 180, 0]
+    ustar = FT(0.28) # 0.28 is taken from Bomex. TODO: Approximate from LES TKE.
+    thermo_params = CAP.thermodynamics_params(params)
+    ts = TD.PhaseNonEquil_pθq(thermo_params, p, θ, TD.PhasePartition(q_vap))
+    T = TD.air_temperature(thermo_params, ts)
+    shf = FT.(Dierckx.Spline1D(t_data, shf_data; k = 1))
+    lhf = FT.(Dierckx.Spline1D(t_data, lhf_data; k = 1))
+    # TODO: Replace Dierckx with a type-stable interpolation and remove the FT.
+    function surface_state(surface_coordinates, interior_z, t)
+        fluxes = HeatFluxes(; shf = shf(t), lhf = lhf(t))
+        parameterization = MoninObukhov(z0; fluxes, ustar)
+        return SurfaceState(; parameterization, T, p, q_vap)
+    end
+    return surface_state
+end
+
+struct DYCOMS_RF01 end
+function (::DYCOMS_RF01)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(15)
+    lhf = FT(115)
+    parameterization = MoninObukhov(z0; fluxes = HeatFluxes(; shf, lhf))
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct DYCOMS_RF02 end
+function (::DYCOMS_RF02)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(16)
+    lhf = FT(93)
+    ustar = FT(0.25)
+    parameterization = MoninObukhov(z0; fluxes = HeatFluxes(; shf, lhf), ustar)
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct Rico end
+function (::Rico)(params)
+    FT = eltype(params)
+    thermo_params = CAP.thermodynamics_params(params)
+    T_surface = FT(299.8)
+    p_surface = FT(101540)
+    z0 = FT(1.5e-4)
+    Cd = FT(0.001229)
+    Ch = FT(0.001094)
+    Cq = FT(0.001133) # TODO: Add support for Cq to SF.Coefficients.
+    # Saturated surface conditions for a given surface temperature and pressure
+    p_sat_surface =
+        TD.saturation_vapor_pressure(thermo_params, T_surface, TD.Liquid())
+    ϵ_v = TD.Parameters.R_d(thermo_params) / TD.Parameters.R_v(thermo_params)
+    q_surface = ϵ_v * p_sat_surface / (p_surface - p_sat_surface * (1 - ϵ_v))
+
+    function surface_state(surface_coordinates, interior_z, t)
+        # Adjust the coefficients from 20 m to the actual value of z.
+        adjustment = (log(20 / z0) / log(interior_z[1, 1, 1, 1, 1] / z0))^2
```

Comment:
```suggestion
        adjustment = (log(20 / z0) / log(interior_z / z0))^2
```
You shouldn't need the `[1, 1, 1, 1, 1]` because `surface_state` should get broadcasted over the surface coordinates.